go/vt/vtgate/planbuilder/operator_transformers.go

```diff
@@ -55,6 +55,29 @@ func transformToLogicalPlan(ctx *plancontext.PlanningContext, op abstract.Physic
 		return transformCorrelatedSubQueryPlan(ctx, op)
 	case *physical.Derived:
 		return transformDerivedPlan(ctx, op)
+	case *physical.Filter:
+		plan, err := transformToLogicalPlan(ctx, op.Source)
+		if err != nil {
+			return nil, err
+		}
+		scl := &simpleConverterLookup{
+			canPushProjection: true,
+			ctx:               ctx,
+			plan:              plan,
+		}
+		ast := sqlparser.AndExpressions(op.Predicates...)
+		predicate, err := evalengine.Convert(ast, scl)
+		if err != nil {
```

Comment:
Should we throw a different error here after we see an error here. One similar to we used to have earlier - "unsupported: cross-shard left join and where clause"

Reply:
Yeah, I was thinking the same